apps/cowswap-frontend/src/modules/swap/hooks/useCheckEthFlowOrderExists.ts

```diff
@@ -0,0 +1,39 @@
+import { useAtomValue } from 'jotai'
+import { useCallback } from 'react'
+
+import { ZERO_ADDRESS } from 'legacy/constants/misc'
+import { useEthFlowContract } from 'legacy/hooks/useContract'
+
+import { ethFlowInFlightOrderIdsAtom } from '../state/EthFlow/ethFlowInFlightOrderIdsAtom'
+
+export interface EthFlowOrderExistsCallback {
+  (orderId: string, orderDigest: string): Promise<boolean>
+}
+
+export function useCheckEthFlowOrderExists(): EthFlowOrderExistsCallback {
+  const ethFlowInFlightOrderIds = useAtomValue(ethFlowInFlightOrderIdsAtom)
+  const ethFlowContract = useEthFlowContract()
+
+  return useCallback(
+    async (orderId: string, orderDigest: string) => {
+      if (ethFlowInFlightOrderIds.includes(orderId)) {
+        return true
+      }
```

Comment:
cool, it still uses the inFlight ids :) 

this will make it very unlikely, but at the same time we will do it when we are unsure. Thats great!